AutoRest/Generators/Ruby/Azure.Ruby/TemplateModels/PageTemplateModel.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Rest.Generator.ClientModel;
+using Microsoft.Rest.Generator.Utilities;
+using Microsoft.Rest.Generator.Ruby;
+using Microsoft.Rest.Generator.Azure.Ruby.Templates;
+
+namespace Microsoft.Rest.Generator.Azure.Ruby
+{
+    public class PageTemplateModel : ModelTemplateModel
+    {
+        public PageTemplateModel(CompositeType source, ISet<CompositeType> allTypes, string nextLinkName, string itemName)
+            : base(source, allTypes)
+        {
+            this.NextLinkName = nextLinkName;
+            this.ItemName = itemName;
+            this.NextLinkParameter = BuildNextLinkParameter();
+        }
+
+        public string NextLinkName { get; private set; }
+
+        public Parameter NextLinkParameter { get; private set; }
+
+        public string ItemName { get; private set; }
+
+        public CompositeType ItemType
+        {
+            get
+            {
+                if (Properties == null)
+                {
+                    return null;
+                }
+                var property = Properties.FirstOrDefault(p => p.Type is SequenceType);
+                if (property != null)
+                {
+                    return ((SequenceType)property.Type).ElementType as CompositeType;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+
+        public Parameter BuildNextLinkParameter()
+        {
+            Parameter param = new Parameter
+            {
+                Name = RubyCodeNamer.RemoveInvalidCharacters(RubyCodeNamer.UnderscoreCase(NextLinkName)),
+                SerializedName = NextLinkName,
+                Type = new PrimaryType(KnownPrimaryType.String),
+                Documentation = "The NextLink from the previous successful call to List operation.",
+                IsRequired = true,
+                Location = ParameterLocation.Path
+            };
+            param.Extensions[Generator.Extensions.SkipUrlEncodingExtension] = true;
+            return param;
+        }
+
+        /// <summary>
+        /// Returns code for declaring inheritance.
+        /// </summary>
+        /// <returns>Code for declaring inheritance.</returns>
+        public override string GetBaseTypeName()
+        {
+            if (this.BaseModelType != null)
+            {
+                string typeName = this.BaseModelType.Name;
+
+                if (this.BaseModelType.Extensions.ContainsKey(AzureExtensions.ExternalExtension) ||
```

Comment:
Could you please elaborate why we may try to find external resource in MsRestAzure:: namespace as the condition also includes AzureExtensions.ExternalExtension